Add warning suppression(rawtypes) to generated CustomType#javaType method

xrd opened the pull request:
<!--**Pull Request Labels**

While not necessary, you can help organize our pull requests by labeling this issue when you open it.  To add a label automatically, simply [x] mark the appropriate box below:

- [ ] feature
- [ ] blocking
- [ ] docs

To add a label not listed above, simply place `/label another-label-name` on a line by itself.
-->

When using javac with warnings as errors, CustomTypes fail compilation with rawtypes error. This adds  a switch to the apollo plugin which can add the proper annotation (`@SuppressWarning("rawtypes")`) to the generated custom type file.

xrd commented:
Summary is at issue #926, and sample code demonstrating the error is here: https://github.com/xrd/apollo_custom_types_error

sav007 commented:
Thanks for the PR. There is one failing test `JavaTypeResolverTest `. Plus I think that you should add tests for compiler, look at https://github.com/apollographql/apollo-android/tree/master/apollo-compiler/src/test/graphql/com/example

To create a new test for compiler you need to:
0. Install the code gen version `npm install -g apollo-codegen@0.19.1`
1.  Create a new folder under `/apollo-compiler/src/test/graphql/com/example`
2. Create `TestOperation.graphql` file with the query
3. Run `./gradlew :apollo-compiler:updateTestIRFiles`
4. Run `com.apollographql.apollo.compiler.CodeGenTest` unit tests
5. Put generate java files from `/apollo-compiler/build/generated/source/apollo/com/example/your-folder` to `/apollo-compiler/src/test/graphql/com/example/your-folder`
6. Run again `com.apollographql.apollo.compiler.CodeGenTest` to verify all tests are green

xrd commented:
@sav007 Great, thanks for the help. I believe I have fixed the failing test and added another which verifies my compiler output is correct when the `useRawTypesWarningSuppression` apollo plugin switch is provided.

digitalbuddha commented on README.md:
```diff
@@ -211,14 +211,14 @@ ApolloClient.builder()
 apollo {
     customTypeMapping['URL'] = "java.lang.String"
-    useRawTypesSuppression = "true"
+    useRawTypesWarningSuppression = "true"
```
Nitpick: `supressRawTypesWarning` sounds better imo

xrd replied:
👍

xrd commented:
@digitalbuddha and @sav007 do I need to do anything else? Not in a rush to get this merged, but let me know if there is anything needed on my end.

hwillson commented on apollo-compiler/src/main/kotlin/com/apollographql/apollo/compiler/Annotations.kt:
```diff
@@ -12,5 +12,7 @@ object Annotations {
   val GENERATED_BY_APOLLO: AnnotationSpec = AnnotationSpec.builder(Generated::class.java)
       .addMember("value", CodeBlock.of("\$S", "Apollo GraphQL")).build()
+  val SUPPRESS_WARNINGS: AnnotationSpec = AnnotationSpec.builder(SuppressWarnings::class.java)
+          .addMember( "value", CodeBlock.of( "\$S", "rawtypes")).build()
```
Can you adjust the indentation here to line up with the rest of the file?

hwillson commented on apollo-compiler/src/main/kotlin/com/apollographql/apollo/compiler/CustomEnumTypeSpecBuilder.kt:
```diff
@@ -37,10 +37,15 @@ class CustomEnumTypeSpecBuilder(
+                  }
+                .addAnnotation(Override::class.java)
+                .returns(Class::class.java)
+                .addStatement("return \$T.class", javaTypeName.toJavaType())
```
The indentation seems to be off here as well - can you adjust it to match the rest of the file?

hwillson commented on apollo-compiler/src/test/graphql/com/example/custom_scalar_type_warnings/custom_scalar_type_warnings/TestQuery.java:
```diff
@@ -0,0 +1,169 @@
+public final class TestQuery implements Query<TestQuery.Data, Optional<TestQuery.Data>, Operation.Variables> {
+  public static final String OPERATION_DEFINITION = "query TestQuery {\n"
+      + "  links\n"
+      + "}";
```
Please keep indentation to 2 space here.

xrd commented:
@hwillson I've pushed the changes to fix the formatting. Thanks!
